File: python/数组/1636. 按照频率将数组升序排序.py
from typing import List
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        from collections import Counter
        d = dict(Counter(nums))
        tmp = sorted(d.items(),key=lambda x:(x[1],-x[0]))
        print(tmp)
        res = []
        for i,j in tmp:
            res+=[i]*j
        return res

s = Solution()
nums = [2,3,1,3,2]
print(s.frequencySort(nums))

File: go/数组/1636. 按照频率将数组升序排序.go
package array

import (
	"sort"
)

func frequencySort(nums []int) []int {
	hash := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		hash[nums[i]]++
	}
	sort.Slice(nums, func(i, j int) bool {
		if hash[nums[i]] == hash[nums[j]] {
			return nums[i] > nums[j]
		}
		return hash[nums[i]] < hash[nums[j]]
	})
	return nums
}
